tmt/utils.py

```diff
@@ -2034,9 +2034,13 @@ def dict_to_yaml(
     def _represent_path(representer: Representer, data: Path) -> Any:
         return representer.represent_scalar('tag:yaml.org,2002:str', str(data))
 
+    def _represent_shell_script(representer: Representer, data: ShellScript) -> Any:
+        return representer.represent_scalar('tag:yaml.org,2002:str', str(data))
+
     yaml.representer.add_representer(pathlib.Path, _represent_path)
     yaml.representer.add_representer(pathlib.PosixPath, _represent_path)
     yaml.representer.add_representer(Path, _represent_path)
+    yaml.representer.add_representer(ShellScript, _represent_shell_script)
```

Comment:
I still wonder whether this is really needed. Where do we save `ShellScript` now where we did not save them before? There are two new fields in `Test` class, true, but there was already `Test.test` which is `Optional[ShellScript]` already, so why your addition requires a representer?

Reply:
Maybe I found it. `Test.test` works thanks to [this](https://github.com/teemtee/tmt/blob/atomasov-setup-cleanup-test/tmt/base.py#L1102C16-L1102C16):
```python
            elif key == 'test' and isinstance(value, ShellScript):  # noqa: SIM114
                data[key] = str(value)
```

So I can add similar `elif` for `setup` and `cleanup`. But I think the yaml representer is a better solution, and it can also handle `Test.test`.

Currently, the `Test._export()` without representer looks like this: 
```
 'test': './runtest.sh',
 'path': '/default-0/tests/tests',
 'setup': [<tmt.utils.ShellScript at 0x7f967858f550>],
 'cleanup': [<tmt.utils.ShellScript at 0x7f967858f5d0>],
```

Reply:
I see, right, that makes sense. But, the output of `_export()` is not necessarily going to meet `dict_to_yaml`. There are other plugins consuming it, e.g. one that renders a Jinja2 template or dumps it as a JSON blob, and these all rely on the output of `_export()` consisting of base Python data types. Therefore I'd advise against the custom representer here, adding it would fix just one use of `_export` output.

Reply:
It should be fixed.